### DISCUSSION TITLE
MacCamy & Fuchs diffraction model implementaion

### DISCUSSION THREAD

**Discussion by MSaraber** [OpenFAST/openfast#2109] [Model Integration]
Hi, 

I have a question regarding the MacCamy & Fuchs diffraction model implementation. I see in this document: https://www.nrel.gov/docs/fy23osti/83679.pdf that it has been implemented in OpenFAST. However, I cant seem to find it in the HydroDyn module. Has this MacCamy & Fuchs correction been made public? Are there other ways to implement the diffraction for a 22MW monopile. I can use potential theory, but this also has its flaws.. 

Thanks in advance.

**Comment by jjonkman** [comment]
Dear @MSaraber,

The MacCamy & Fuchs diffraction model has been implemented in HydroDyn via the following pull request: https://github.com/OpenFAST/openfast/pull/1008.  However, this functionality is not included in the main branch of OpenFAST or in the current release of OpenFAST v3.5.2.  This functionality is slated for release in the upcoming OpenFAST v4.0 release.

For a fixed-bottom monopile, I would generally recommend to use the strip-theory solution (with the MacCamy & Fuchs diffraction model for large-diameter monopiles) rather than the potential-flow solution in HydroDyn.

Best regards,

**Comment by MSaraber** [comment]
Thank you for your response @jjonkman. I have compiled the dev branch of OpenFAST. I was wondering if there is a template of the .fst file and module files. For example: I can't seem to find the SeaState module and how the .fst file changes to run the dev branch of OpenFAST. Thanks in advance again.

**Comment by luwang00** [comment]
Hi @MSaraber,

We have some example input files compatible with OpenFAST 4.0 on the [`dev-unstable-pointers` branch of our r-test repository](https://github.com/OpenFAST/r-test/tree/dev-unstable-pointers). 

You can also find some examples making use of the MacCamy & Fuchs diffraction model with the various wave-stretching model added to 4.0 [here](https://github.com/OpenFAST/r-test/tree/dev-unstable-pointers/modules/hydrodyn). These are the cases with "MCF" in the name.

In the SeaState input file, there's an entry called `MCFD`. This needs to be the same as the monopile member diameter in the HydroDyn input file. In the HydroDyn input file, all `Cp` entries assigned to the monopile member using the MacCamy & Fuchs diffraction model should be set to `MCF` instead of a numerical value. Please refer to the examples linked above.

I would also suggest compiling the `dev-unstable-pointers` branch of OpenFAST instead of the `dev` branch to get a build of 4.0. I think the former is more up to date.

**Reply by MSaraber** [reply]
Hi @luwang00, 

Thank you very much for your response and pointing me toward the examples. I will be sure to compile the dev-unstable-pointers branch.
